nodes/geometry/intersect_polyline_plane.py

```diff
@@ -0,0 +1,130 @@
+import bpy
+from bpy.props import *
+from mathutils import Vector, geometry
+from ... events import executionCodeChanged
+from ... base_types.node import AnimationNode
+
+edgesTypeItems = [  ("POINTS", "Points in order", ""),
+                    ("EDGES", "Points by edges", "") ]
+
+class IntersectPolylinePlaneNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_IntersectPolylinePlaneNode"
+    bl_label = "Intersect Polyline Plane"
+    bl_width_default = 160
+    
+    def edgesTypeChanged(self, context):
+        executionCodeChanged()
+    
+    edgesType = EnumProperty(name = "Plane Type", default = "POINTS",
+        items = edgesTypeItems, update = edgesTypeChanged)
+    message = StringProperty(name = "Message", default = "Expecting Points")
+    cyclic = BoolProperty(name = "Cyclic points", description = "Consider last point to first point also", 
+        default = True, update = executionCodeChanged)
+        
+    def create(self):
+        self.newInput("Vector", "Positions", "positions")
+        self.newInput("Edge Indices List", "Edge Indices", "edges")
```

Comment:
the convention I use everywhere is:
socket = self.newInput(...)
socket.setting1 = ...
socket.setting2 = ...